### ISSUE TITLE
question: how to share python versions with colleagues

### ISSUE THREAD

**Issue by emacs18** [pyenv/pyenv#1925]
I use CentOS 6  linux servers.  Many dozens if not hundreds of my colleagues also use linux servers all of which use the same network file system.

How can I allow my colleagues to use my build of different python so that they don't have to build python themselves?
I built few python 2 and 3 versions.  It was very painful which required me to install even things like openssl.  I would like to alleviate the pain from my colleagues.  For example could it be as simple as creating symlinks like this:

```
cd ~/.pyenv/versions
ln -s ~pyenvguru/.pyenv/versions/2.7.18
ln -s ~pyenvguru/.pyenv/versions/3.10-dev
```

where `pyenvguru` is the user who has installed python versions?

Same question with virtual environments.  Could these also be shared among users?

What I would like is to setup a set of python versions as well as few virtual environments.  I would like  to have this available to my colleagues so that they can leverage my work without having to repeat them.  On modern OS versions such as ubuntu 20.04, it is easy enough that there should be no need to share anything among users.  However it takes more work for those of stuck on ancient OS.  Many of my colleagues would not be able to use pyenv without help.

**Comment by native-api** [comment]
Yes, you can link to installed Python versions and even install Pyenv system-wide, e.g. into `/opt` (e.g. that's what Travis CI does to make alternative Python versions available).

However, since Python links to dependencies dynamically, the installations may not work on other systems which lack them or have different versions.

**Comment by native-api** [comment]
> However, since Python links to dependencies dynamically, the installations may not work on other systems which lack them or have different versions.

You can work around that [by passing custom build flags to the Python build](https://github.com/pyenv/pyenv/tree/master/plugins/python-build#special-environment-variables) -- e.g. to have it link to the dependencies statically.

**Comment by emacs18** [comment]
Thanks for your answers.  I'm not sure what you mean by "systems which lack them".  In my case all linux servers have access to same networked files, i.e., all shared objects that my build of python may depend on will also be available to all linux servers.  So no linux server should be missing any shared object that my build of python may have depended on. 

My hope is that only I need to know how to pass build flags to build python so that all my colleagues can simply use them.

**Comment by native-api** [comment]
Libraries are typically installed into system locations (e.g. `/usr/lib` or `/usr/local/lib`) which are not available to other machines.

But you install then in such a way that they are -- all the better for you!

**Comment by native-api** [comment]
Libraries are typically installed into system locations (e.g. `/usr/lib` or `/usr/local/lib`) which are not available to other machines.

But you're installing them in such a way that they are -- all the better for you!

**Comment by emacs18** [comment]
First all servers are imaged from same image so that /usr/lib64 should be the same.
I don't have root access, so I cannot write anything under any local disks.
I installed all other dependent libraries such as openssl on a networked file system access-able to all servers.

So I'm going to have few of my colleagues start testing pyenv by using symlinks.
I plan on documenting all the things I needed to do to get pyenv working on CentOS 6.
If such note may be useful to others, I can upload that document somewhere if you wish.

Thanks for your help.
